src/test/java/com/google/api/codegen/testdata/nodejs_test_library.baseline

```diff
@@ -0,0 +1,719 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+
+var assert = require('assert');
+var service = require('../src/').v1();
+
+describe('LibraryServiceClient', function() {
+  describe('createShelf', function() {
+    it('createShelfTest', function(done) {
+      var client = service.LibraryServiceClient();
+      // Mock request
+      var shelf = {};
+      var request = {
+          shelf : shelf
+      };
+
+      // Mock response
+      var name = 'name3373707';
+      var theme = 'theme110327241';
+      var internalTheme = 'internalTheme792518087';
+      var expectedResponse = {
+          name : name,
+          theme : theme,
+          internalTheme : internalTheme
+      };
+
+      // Mock Grpc layer
+      client._createShelf = function(actualRequest, options, callback) {
+        assert.equal(actualRequest.shelf, shelf);
+        callback(null, expectedResponse);
+      };
+
+      client.createShelf(request, function(err, response) {
+        assert.ifError(err);
+        assert.equals(response, expectedResponse);
+        done();
+      });
+    });
+  };
+
+  describe('getShelf', function() {
+    it('getShelfTest', function(done) {
+      var client = service.LibraryServiceClient();
+      // Mock request
+      var formattedName = client.shelfPath("[SHELF_ID]");
+      var options = 'options-1249474914';
+      var request = {
+          name : formattedName,
+          options : options
+      };
+
+      // Mock response
+      var formattedName2 = client.shelfPath("[SHELF_ID]");
+      var theme = 'theme110327241';
+      var internalTheme = 'internalTheme792518087';
+      var expectedResponse = {
+          name : formattedName2,
+          theme : theme,
+          internalTheme : internalTheme
+      };
+
+      // Mock Grpc layer
+      client._getShelf = function(actualRequest, options, callback) {
+        assert.equal(actualRequest.name, formattedName);
+        assert.equal(actualRequest.options, options);
+        callback(null, expectedResponse);
+      };
+
+      client.getShelf(request, function(err, response) {
+        assert.ifError(err);
+        assert.equals(response, expectedResponse);
+        done();
+      });
+    });
+  };
+
+  describe('listShelves', function() {
+    it('listShelvesTest', function(done) {
+      var client = service.LibraryServiceClient();
+      // Mock request
+      var request = {};
+
+      // Mock response
+      var nextPageToken = '';
+      var shelvesElement = {};
+      var shelves = [shelvesElement];
+      var expectedResponse = {
+          nextPageToken : nextPageToken,
+          shelves : shelves
+      };
+
+      // Mock Grpc layer
+      client._listShelves = function(actualRequest, options, callback) {
+        callback(null, expectedResponse, expectedResponse.nextPageToken);
+      };
+
+      client.listShelves(request, function(err, response, nextPageToken) {
```

Comment:
This is not a normal style for paged iterations in Node.JS.  By default, only `response` should exist, which is an array of the response. 

When called as `listShelves(request, {autoPaginate: false}, function(err, response, next) {....})`, the `response` will be an array in a page, and `next` will be the request object which can be used to fetch the next page. It would be nicer to respect the same style.